packages/website/ts/components/portal/portal.tsx

```diff
@@ -230,14 +244,29 @@ export class Portal extends React.Component<PortalProps, PortalState> {
     private _onTokenChosen(tokenAddress: string) {
         if (_.isEmpty(tokenAddress)) {
             this.setState({
-                isAssetPickerDialogOpen: false,
+                tokenManagementState: TokenManagementState.None,
             });
             return;
         }
         const token = this.props.tokenByAddress[tokenAddress];
-        this.props.dispatcher.updateTokenByAddress([token]);
+        const isDefaultTrackedToken = _.includes(configs.DEFAULT_TRACKED_TOKEN_SYMBOLS, token.symbol);
+        if (this.state.tokenManagementState === TokenManagementState.Remove && !isDefaultTrackedToken) {
+            if (token.isRegistered) {
+                // Remove the token from tracked tokens
+                const newToken = {
+                    ...token,
+                    isTracked: false,
+                };
+                this.props.dispatcher.updateTokenByAddress([newToken]);
+            } else {
+                this.props.dispatcher.removeTokenToTokenByAddress(token);
+            }
+            trackedTokenStorage.removeTrackedToken(this.props.userAddress, this.props.networkId, tokenAddress);
+        } else if (isDefaultTrackedToken) {
+            this.props.dispatcher.showFlashMessage(`Cannot remove ${token.name} because it's a default token`);
+        }
```

Comment:
Where do we handle `TokenManagementState.Add`?

Reply:
Ah, I see, maybe just add a TODO comment here?

Reply:
seems that the AssetPicker component will automatically track the token